src/vs/editor/contrib/linesOperations/common/linesOperations.ts

```diff
@@ -726,6 +733,23 @@ export class TransposeAction extends EditorAction {
 		editor.executeCommands(this.id, commands);
 		editor.pushUndoStop();
 	}
+
+	_runTransposeSelections(accessor: ServicesAccessor, editor: ICommonCodeEditor): void {
+		let selections = editor.getSelections();
+		let model = editor.getModel();
+		let commands: ICommand[] = [];
+		let selectionText = selections.map(selection => model.getValueInRange(selection));
+
+		for (let i = 0, len = selections.length; i < len; i++) {
+			let selection = selections[i];
+			let replacementText = selectionText[(i - 1 + selectionText.length) % selectionText.length];
+			commands.push(new ReplaceCommandThatSelectsReplacement(selection, replacementText));
```

Comment:
we can directly use `EditOperation.replace` and run `editor.executeEdits` at the end instead of composing commands.